codec-dns/src/main/java/io/netty/handler/codec/dns/TcpDnsQueryDecoder.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2019 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.handler.codec.dns;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.handler.codec.CorruptedFrameException;
+import io.netty.handler.codec.LengthFieldBasedFrameDecoder;
+import io.netty.util.internal.ObjectUtil;
+
+public final class TcpDnsQueryDecoder extends LengthFieldBasedFrameDecoder {
+    private final DnsRecordDecoder decoder;
+
+    public TcpDnsQueryDecoder() {
+        this(DnsRecordDecoder.DEFAULT, Integer.MAX_VALUE);
```

Comment:
DNS Packets cannot be greater than `65,535` bytes. I think we should change `Integer#MAX_VALUE` with `65,535`.

Reply:
Changed to `unsigned short max value 65535`